azurerm/internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -807,6 +822,13 @@ func resourceKubernetesClusterCreate(d *schema.ResourceData, meta interface{}) e
 		}
 	}
 
+	if v, ok := d.GetOk("private_dns_zone_id"); ok {
+		if parameters.Identity == nil || parameters.Identity.Type != containerservice.ResourceIdentityTypeUserAssigned {
```

Comment:
this likely only applies when not using "System" - could we add an acctest for this?